### PR TITLE
Remove unused variables

### PR DESCRIPTION
**Pull request by NanoSector**
Scrutinizer reports about 400 instances of unused variables and unused code in general. This PR should help with that.

### CODE REVIEW

**Comment by NanoSector** [comment]
So the PR merged just now caused a merge conflict and #3763 will too. Not sure how to solve this.

**Comment by colinschoen** [comment]
Manually merge master into your branch via the command line and fix the conflicts. I can do this for you if you aren't sure how.

**Comment by Sesquipedalian** [comment]
> So the PR merged just now caused a merge conflict and #3763 will too. Not sure how to solve this.

Once #3763 is merged into `release-2.1` on the official repo, do the following:

1. Check out your local copy of `release-2.1`
2. Pull in the latest changes from origin
3. Check out your `randombugfixes` branch
4. Run `git merge release-2.1`
5. Edit the conflicting files to resolve the conflicts
6. Run `git commit -as` to add and commit (and sign) your conflict resolutions.
7. Push to this PR

**Comment by NanoSector** [comment]
I don't know what I'm doing right here, haha. I'll wait for the other PR to be merged.

**Comment by live627** [comment]
I can rebase things for you if you'd like.

**Comment by NanoSector** [comment]
I think I figured it out, thanks

**Comment by live627** [comment]
ah cool

nice to see you back btw!

**Comment by NanoSector** [comment]
I'm glad to be back :) Forgot how much fun SMF was.

**Comment by live627** [comment]
The entire branch now seems duplicated.I did a rebase https://github.com/SimpleMachines/SMF2.1/compare/release-2.1...live627:randombugfixes

should I push that to this PR?

**Comment by NanoSector** [comment]
Thanks! Not sure what went wrong.

File: Sources/ManageLanguages.php
Comment (Oldiesmann) [inline Sources/ManageLanguages.php]: All of this was removed in #3760.

File: Sources/Subs-Db-postgresql.php
Comment (Oldiesmann) [inline Sources/Subs-Db-postgresql.php:493]: Is there a reason for removing this check? $connection in this case is a DB connection resource. It could possibly be false if the connection failed, in which case defaulting to $db_connection is probably wise.

**Review by Oldiesmann — CHANGES_REQUESTED** [review]
Looks good. Just a couple of minor issues that need to be addressed.

File: Sources/ManageLanguages.php
Comment (NanoSector) [reply Sources/ManageLanguages.php]: I tried merging this but fucked up, @live627's branch might have this fixed.

File: Sources/Subs-Db-postgresql.php
Comment (NanoSector) [reply Sources/Subs-Db-postgresql.php:493]: $connection and therefore $db_connection is not used anywhere in that function.

File: Sources/ManageLanguages.php
Comment (NanoSector) [reply Sources/ManageLanguages.php]: Yeah: https://github.com/SimpleMachines/SMF2.1/commit/b13e4cc16c93c92de4ad54d39230e25696e5d3d9

File: Sources/Subs-Db-postgresql.php
Comment (Oldiesmann) [inline Sources/Subs-Db-postgresql.php]: $connection is passed as a parameter to that function.

Comment (Oldiesmann) [reply Sources/Subs-Db-postgresql.php:493]: Ah. Didn't notice that. Nevermind then :smile:

**Review by Oldiesmann — APPROVED** [review]

**Pull request merged**
